lib/mayaUsd/fileio/writeJobContext.cpp

```diff
@@ -63,8 +63,14 @@ namespace {
 
 inline SdfPath _GetRootOverridePath(const UsdMayaJobExportArgs& args, const SdfPath& path)
 {
-    if (!args.usdModelRootOverridePath.IsEmpty() && !path.IsEmpty()) {
-        return path.ReplacePrefix(path.GetPrefixes()[0], args.usdModelRootOverridePath);
+    if (!path.IsEmpty()) {
+        if (!args.usdModelRootOverridePath.IsEmpty()) {
+            return path.ReplacePrefix(path.GetPrefixes()[0], args.usdModelRootOverridePath);
+        }
+
+        if (!args.rootMapFunction.IsNull()) {
+            return args.rootMapFunction.MapSourceToTarget(path);
+        }
```

Comment:
It is not clear to me that these two options are mutually exclusive. Can there be a scenario where there are both root overrides and export roots?

Reply:
Good point. I would hope we will unify both rootMapFunction and usdModelRootOverridePath. It's outside of the scope for this PR, so I will abort when both are set and issue a message to the user - 6d352eb17ebb129804aa16364c33fffd84f83642

Reply:
I think we'd better put some restrictions on how these flags can be mingled. The `usdModelRootOverridePath` is not a command argument itself, and that means it has pretty restricted use and is only used in the `renderLayerMode==modelingVariant` case. It's a pretty neat workflow for authoring variants but would probably be used mostly in the modeling phase which shouldn't need to use export roots too extensively since they are the beginning of the pipeline and can control their parent groups better. In my quick test, it seemed like it only exports one top level hierarchy anyway even if two are selected (maybe that should error?).

This is what I'd propose:
a) support using both flags if there is only one export root. I believe you could change this function to do the remap first and then the replacement. We should check for other places doing that `ReplacePrefix()`.
b) explicitly forbid using `-ert` in render layer mode until someone asks for it.

A caveat here is that we currently don't use the modelingVariant mode (which is probably why we wrote it like this and haven't suffered). Though it may be a cool feature for modelers, I've been a bit weary to jump on it if its still using legacy render layers and if there may be other more awesome variant authoring abilities in the future. I think a) strikes a nice balance between the alternatives.